optuna/integration/shap.py

```diff
@@ -0,0 +1,86 @@
+from collections import OrderedDict
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import numpy as np
+
+from optuna._imports import try_import
+from optuna.importance._base import BaseImportanceEvaluator
+from optuna.importance._mean_decrease_impurity import MeanDecreaseImpurityImportanceEvaluator
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+
+
+with try_import() as _imports:
+    from shap import TreeExplainer
+
+
```

Comment:
Could you add `experimental`? 
```suggestion

@experimental("3.0.0")
```

Note that we need to add `from optuna._experimental import experimental` to the import block